### PR TITLE
feat(filter-field): Prefills text of first level freeText node on edit

### PR DESCRIPTION
**Pull request by rowa-audil**
APM-309830

### <strong>Pull Request</strong>

<hr>
Hi, thank you for contributing to Barista with this pull request (PR).

To ensure a fast process and merging of your PR please make sure it fulfills the
coding standards and contribution guidelines.

- A feature proposal has been provided, discussed and approved first.
- There is a meaningful description of the issue in GitHub (Screenshots are
  often helpful).
- If the PR introduces breaking-changes or deprecations it matches the following
  guidelines.
  - The commit message follows our commit guidelines.
  - Tests for the changes have been added (for bug fixes / features).
  - Docs have been added / updated (for bug fixes / features).

Please choose the type appropriate for the changes below: <br>

#### Checklist

- [ ] I have read the CONTRIBUTING doc and I follow the PR guidelines
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: libs/barista-components/filter-field/src/filter-field.ts [hunk 1]
```diff
@@ -1119,6 +1119,10 @@ export class DtFilterField<T = any>
               recentRangeValue.operator as DtFilterFieldRangeOperator,
             );
           }
+          if (isDtFreeTextDef(this._currentDef)) {
+            this._inputValue = recentRangeValue.toString();
```
Comment (thomaspink) [inline libs/barista-components/filter-field/src/filter-field.ts:1123]: Not sure what the `recentRangeValue` has to do with a `free-text` - can you clarify it for me?

File: libs/barista-components/filter-field/src/filter-field.ts [hunk 2]
```diff
@@ -1119,6 +1119,10 @@ export class DtFilterField<T = any>
               recentRangeValue.operator as DtFilterFieldRangeOperator,
             );
           }
+          if (isDtFreeTextDef(this._currentDef)) {
```
Comment (thomaspink) [inline libs/barista-components/filter-field/src/filter-field.ts:1122]: I am missing the check if we are on a 2nd level free-text. Can you point me in the right direction?

**Review by thomaspink — CHANGES_REQUESTED** [review]
Also a test would be great for that use-case

File: libs/barista-components/filter-field/src/filter-field.ts (hunk 2 above)
Comment (rowa-audil) [reply libs/barista-components/filter-field/src/filter-field.ts:1122]: Offline, I've discussed with Tom how the free-text editting is handled. Second layer and lower will reset the filter value upon edit to the first layer, resulting in the free-text not existing in the menu. This reason removes the necessity for second or lower layer support for free-text-edit-prefills

File: libs/barista-components/filter-field/src/filter-field.ts (hunk 1 above)
Comment (rowa-audil) [reply libs/barista-components/filter-field/src/filter-field.ts]: Renamed to initialRecentlyRemoved

Comment (thomaspink) [reply libs/barista-components/filter-field/src/filter-field.ts]: This does not solve the issue that we want to prefill free-text not ranges! You are reading the range value from the removed ones an setting it into the free-text input. This does not make sense

Comment (rowa-audil) [reply libs/barista-components/filter-field/src/filter-field.ts]: The check at L1122 tells me that the current definition is a free-text, so I'm rest assured that the removed item is the string to prefill the input

Comment (thomaspink) [reply libs/barista-components/filter-field/src/filter-field.ts]: got it. thank you

**Review by thomaspink — DISMISSED** [review]

File: libs/barista-components/filter-field/src/filter-field-edit-mode.spec.ts [hunk 3]
```diff
@@ -368,7 +379,7 @@ describe('DtFilterField', () => {
 
       // Read the filters again and make expectations
       const filterTags = getFilterTags(fixture);
-
+      console.log('FILTER TAGS: ', filterTags);
```
Comment (tomheller) [inline libs/barista-components/filter-field/src/filter-field-edit-mode.spec.ts:382]:
```suggestion

```

File: libs/barista-components/filter-field/src/filter-field-edit-mode.spec.ts [hunk 4]
```diff
@@ -462,9 +473,33 @@ describe('DtFilterField', () => {
...
+    it('should not prefill the input on edit of a second layer free-text filter', () => {
+      const tags = fixture.debugElement.queryAll(
+        By.css('.dt-filter-field-tag-label'),
+      );
+      tags[3].nativeElement.click();
+      advanceFilterfieldCycle();
+
+      const inputField = getInput(fixture);
+
+      expect(inputField.value).not.toBe('Custom free text');
```
Comment (tomheller) [inline libs/barista-components/filter-field/src/filter-field-edit-mode.spec.ts:502]: Can we have a bit clearer expectation here. There are a lot of things in the world that are `not` _Custom free text_.

What do we expect here?

File: libs/examples/src/filter-field/filter-field-default-search-example/filter-field-default-search-example.ts [hunk 5]
```diff
@@ -35,11 +35,7 @@ export class DtExampleFilterFieldDefaultSearch
       {
         name: 'Asia',
         defaultSearch: true,
-        suggestions: [
-          { name: 'Australia' },
-          { name: 'China' },
-          { name: 'India' },
-        ],
+        suggestions: [{ name: 'China' }, { name: 'India' }],
```
Comment (tomheller) [inline libs/examples/src/filter-field/filter-field-default-search-example/filter-field-default-search-example.ts:38]: Why did you remove Australia here?

**Review by tomheller — CHANGES_REQUESTED** [review]

File: libs/barista-components/filter-field/src/filter-field-edit-mode.spec.ts (hunk 3 above)
Comment (ffriedl89) [reply libs/barista-components/filter-field/src/filter-field-edit-mode.spec.ts:382]: its still in there though :D

File: libs/examples/src/filter-field/filter-field-default-search-example/filter-field-default-search-example.ts (hunk 5 above)
Comment (rowa-audil) [reply libs/examples/src/filter-field/filter-field-default-search-example/filter-field-default-search-example.ts:38]: For obvious reasons. Australia belongs to Europe

**Comment by tomheller** [comment]
@rowa-audil Tests are now failing, can you please have a look.

File: libs/examples/src/filter-field/filter-field-default-search-example/filter-field-default-search-example.ts (hunk 5 above)
Comment (tomheller) [reply libs/examples/src/filter-field/filter-field-default-search-example/filter-field-default-search-example.ts:38]: Would you also be able to provide a serious answer?

**Review by tomheller — DISMISSED** [review]

**Review by tomheller — APPROVED** [review]
